app/assets/javascripts/components/amounttable.js

```diff
@@ -0,0 +1,313 @@
+ * @property {function(data: Any): string} [text=null]
+ *           Function that generates the HTML content of the cell
+ *           (html is preferred over text in when rendering a table, text is
+ *           preferred when exporting to csv)
+ * @property {object} [style=null]
+ *           object whose keys will be transformed to CSS properties of the
+ *           cells in the column
+ * @property {function(data: Any): number} [shade=false]
+ *           Function that calculates the amount the shader is filled for cells
+ *           in this column. Should be in [0,100]
+ */
+
+
+/**
+ * A table representation of data
+ */
+class AmountTable {
+    /**
+     * Constructor of an AmountTable
+     *
+     * Pro tip: use `style: {"background-color": "red"}` to change the shader
+     * fill color. Do not use `background` on a shaded cell.
+     *
+     * @param {AmountTableSettings} settings
+     */
+    constructor({el, title=null, header = null, data, contents=null, limit=Infinity, tooltip=null, tooltipID=null}) {
+        this.el = el;
+        this.title = title;
+        this.header = header;
+        this.data = data || [];
+        this.table = null;
+        this.limit = limit;
+        this.collapsed = true;
+        this.contents = contents;
+        this.makeTooltip = tooltip;
+        if (tooltip !== null) {
+            if (tooltipID === null) {
+                this.createTooltip();
+            } else {
+                this.tooltip = d3.select(tooltipID);
+            }
+        } else {
+            this.tooltip = null;
+        }
+    }
+
+    /**
+     * Create a tooltip element of class "tip" do display tooltips
+     */
+    createTooltip() {
+        this.tooltip = d3.select("body")
+            .append("div")
+            .attr("class", "tip")
+            .style("position", "absolute")
+            .style("z-index", "10")
+            .style("visibility", "hidden")
+            .style("background-color", "white")
+            .style("padding", "2px")
+            .style("border", "1px solid #dddddd")
+            .style("border-radius", "3px;");
+    }
+
+    /**
+     * Make the table visualization
+     */
+    draw() {
+        const [thead, tbody] = this.buildTable();
+        this.buildHeader(thead);
+        if (this.data.length != 0) {
+            this.buildRow(tbody);
+        } else {
+            // fallback in case there is no data
+            tbody.append("tr").append("td")
+                .attr("colspan", this.header.length)
+                .style("text-align", "center")
+                .text("No data");
+        }
+    }
+
+    /**
+     * Create a table in the target element (without creating it first)
+     * @return {d3.selection}  Table head and table body
+     */
+    buildTable() {
+        if (this.table === null) {
+            this.table = this.el.append("table").attr("class", "table table-condensed table-amounttable");
+        }
+        this.table.html(); // empty the tables HTML
```

Comment:
Without parameter, this returns the html (in both jquery and d3, I don't know what `this.table` is). An empty string should be added as parameter.

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13716) by @bmesuere on Mon Feb 19 2018 at 13:46._